src/matched-data.ts

```diff
@@ -52,7 +55,7 @@ export function matchedData<T extends object = Record<string, any>>(
     .reduce((state, instance) => _.set(state, instance.path, instance.value), {} as T);
 }
 
-function createFieldExtractor(removeOptionals: boolean) {
+function createFieldExtractor(removeOptionals: IncludeOptionals) {
```

Comment:
This one is not a public function so you can change its signature.
I'll explain you why. Private functions are only used from the code inside the package, so we have full control over it. Instead public functions, classes, methods require extra caution when edited because we do not want to introduce a breaking change to someone using the library when we publish a minor version. When we publish a major version we can potentially do whatever we want with breaking changes.